### PR TITLE
add new documents input type

### PR DESCRIPTION
**Pull request by gilgardosh**
Related #7213

### CODE REVIEW

File: packages/graphql-codegen-cli/tests/codegen.spec.ts [hunk 1]
```diff
@@ -928,6 +928,27 @@ describe('Codegen Executor', () => {
         expect(e.message).toContain('Failed to load custom loader');
       }
     });
+
+    it('Should allow loading docs by require+config method', async () => {
+      await executeCodegen({
+        schema: ['./tests/test-documents/schema.graphql'],
+        documents: [
+          {
+            require: './tests/test-documents/valid.graphql',
```
Comment (dotansimha) [inline packages/graphql-codegen-cli/tests/codegen.spec.ts:937]: The goal was actually to do it the other way around. 

So you can use a loader, and pass config to it: 

```
require: ./tests/custom-loaders/custom-documents-loader.js
config:
   loadFilesFrom: "./tests/test-documents/valid.graphql"
```

**Review by dotansimha — CHANGES_REQUESTED** [review]

**Comment by charlypoly** [comment]
@gilgardosh are we good to go? 🚢 
If so, I can take care of rebasing the PR and merging it

File: .changeset/silly-clouds-play.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@graphql-codegen/cli': minor
+'@graphql-codegen/plugin-helpers': patch
+---
+
+Supoport loading documents by require+config structure
```
Comment (charlypoly) [inline .changeset/silly-clouds-play.md:6]:
```suggestion
Support loading documents by require+config structure
```

**Comment by charlypoly** [comment]
@gilgardosh could you confirm if the following looks correct to you? 

https://github.com/dotansimha/graphql-code-generator/pull/7304/files#diff-e51ca8ddc81d655b7024e6d82e2911acd04d919909db7b23d312c1c056feaa49R273

Thanks!

**Comment by gilgardosh** [comment]
@charlypoly Looks great, thank you!

**Comment by charlypoly** [comment]
Discussed with @gilgardosh, will be done in `graphql-tools`.

**Pull request closed without merging**